### PR TITLE
Fix completed scheduled tasks allowed to run multiple times

### LINKED ISSUE #2967: Completed tasks will run a second time in Scheduler
```csharp

var scheduledHide = Scheduler.AddDelayed(() => currentScreen.Hide(), TournamentScreen.FADE_DELAY);

scheduledHide.RunTask();
```

This will run the task twice, as there is no `Completed` check in scheduler. When fixing this, special care should be taken for repeating tasks, which likely should clear their `Completed` status on reschedule.

### PR DESCRIPTION
**Pull request by iiSaLMaN**
Closes #2967 

## Potentially breaking change
Attempting to invoke an already completed task will result in a thrown `InvalidOperationException`.

### CODE REVIEW

**Comment by peppy** [comment]
I'm so confused by the test. why seeking? why custom clock? what are you trying to test here? you aren't asserting that it isn't run before the specified time, and i'm not sure why you are even delaying.

Also path for `RepeatInterval = 0` isn't covered

**Comment by iiSaLMaN** [comment]
> I'm so confused by the test. why seeking? why custom clock? what are you trying to test here? you aren't asserting that it isn't run before the specified time, and i'm not sure why you are even delaying.
> 
> Also path for `RepeatInterval = 0` isn't covered

All concerns should be resolved. sorry for this complicated test, will try to not do it again.

File: osu.Framework.Tests/Threading/SchedulerTest.cs [hunk 1]
```diff
@@ -274,5 +289,28 @@ public void TestScheduleFromInsideDelegate([Values(false, true)] bool forceSched
...
+            ScheduledDelegate del;
+            scheduler.Add(del = new ScheduledDelegate(() => invocations++));
+            Assert.AreEqual(0, invocations);
+
+            scheduler.Update();
+            Assert.AreEqual(1, invocations);
+
+            try
+            {
+                del.RunTask();
```
Comment (peppy) [inline osu.Framework.Tests/Threading/SchedulerTest.cs]:
```suggestion
            Assert.Throws<InvalidOperationException >(del.RunTask);
```

Comment (peppy) [reply osu.Framework.Tests/Threading/SchedulerTest.cs]: no need for try-catch

File: osu.Framework.Tests/Threading/SchedulerTest.cs [hunk 2]
```diff
@@ -291,18 +291,60 @@ public void TestScheduleFromInsideDelegate([Values(false, true)] bool forceSched
         }
 
         [Test]
-        public void TestAttemptInvokeMultipleTimes()
+        public void TestAttemptMultipleInvoke([Values(false, true)] bool afterSchedule)
```
Comment (smoogipoo) [inline osu.Framework.Tests/Threading/SchedulerTest.cs]: Split this out into two tests.

**Pull request merged**